src/System.Windows.Forms/src/System/Windows/Forms/ListViewItem.ListViewSubItem.ListViewSubItemAccessibleObject.cs

```diff
@@ -14,79 +14,85 @@ public partial class ListViewSubItem
 
-                        if (Column == 0 && _owningItem.SubItems.Count > 1)
+                        if (index == 0 && _owningListView.Columns.Count > 1)
                         {
-                            width = _owningItem.SubItems[Column + 1].Bounds.X - _owningSubItem.Bounds.X;
+                            width = _internalParent.GetSubItemBounds(index + 1).X - bounds.X;
+                        }
+
+                        if (width <= 0)
+                        {
+                            return Rectangle.Empty;
                         }
 
                         return new Rectangle(
-                            _owningListView.AccessibilityObject.Bounds.X + _owningSubItem.Bounds.X,
-                            _owningListView.AccessibilityObject.Bounds.Y + _owningSubItem.Bounds.Y,
-                            width, _owningSubItem.Bounds.Height);
+                            _owningListView.AccessibilityObject.Bounds.X + bounds.X,
+                            _owningListView.AccessibilityObject.Bounds.Y + bounds.Y,
+                            width, bounds.Height);
                     }
                 }
 
                 internal override UiaCore.IRawElementProviderFragment? FragmentNavigate(UiaCore.NavigateDirection direction)
-                {
-                    switch (direction)
+                    => direction switch
                     {
-                        case UiaCore.NavigateDirection.Parent:
-                            return _owningItem.AccessibilityObject;
-                        case UiaCore.NavigateDirection.NextSibling:
-                            int nextSubItemIndex = GetCurrentSubItemIndex() + 1;
-                            if (_owningItem.SubItems.Count > nextSubItemIndex)
-                            {
-                                return _owningItem.SubItems[nextSubItemIndex].AccessibilityObject;
-                            }
-
-                            break;
-                        case UiaCore.NavigateDirection.PreviousSibling:
-                            int previousSubItemIndex = GetCurrentSubItemIndex() - 1;
-                            if (previousSubItemIndex >= 0)
-                            {
-                                return _owningItem.SubItems[previousSubItemIndex].AccessibilityObject;
-                            }
-
-                            break;
-                    }
+                        UiaCore.NavigateDirection.Parent
+                            => Parent,
+                        UiaCore.NavigateDirection.NextSibling
+                            => Parent.GetChild(Index + 1),
+                        UiaCore.NavigateDirection.PreviousSibling
+                            => Parent.GetChild(Index - 1),
+                        _ => base.FragmentNavigate(direction)
+                    };
 
-                    return base.FragmentNavigate(direction);
-                }
+                internal override int Column
+                    => Index;
```

Comment:
nit: It takes less than 120 characters
```suggestion
                internal override int Column => Index;
```